diff --git a/src/main/java/org/apache/commons/jexl3/MapContext.java b/src/main/java/org/apache/commons/jexl3/MapContext.java
index a7b85311..da3beeff 100644
--- a/src/main/java/org/apache/commons/jexl3/MapContext.java
+++ b/src/main/java/org/apache/commons/jexl3/MapContext.java
@@ -19,6 +19,7 @@ package org.apache.commons.jexl3;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
 
 /**
  * Wraps a map in a context.
@@ -64,6 +65,16 @@ public class MapContext implements JexlContext {
         return map.containsKey(name);
     }
 
+    /**
+     * Adds all variables from the given map to this context.
+     *
+     * @param variables the variables to add
+     * @throws NullPointerException if variables is null
+     */
+    public void putAll(final Map<String, Object> variables) {
+        map.putAll(Objects.requireNonNull(variables, "variables"));
+    }
+
     @Override
     public void set(final String name, final Object value) {
         map.put(name, value);
